src/CoreBundle/private/library/classes/TCMSResourceCollection.class.php

```diff
@@ -69,98 +81,161 @@ public function IsEnabled()
      */
     public function IsAllowed()
     {
-        $bAllowResourceCollection = false;
-        if ($this->IsEnabled()) {
-            $bAllowResourceCollection = true;
-            $oGlobal = TGlobal::instance();
-            $bIsModuleChooser = $oGlobal->GetUserData('__modulechooser');
-            $bAllowResourceCollection = ($bAllowResourceCollection && !TGlobal::IsCMSMode()); // not in cms backend
-            if ($bAllowResourceCollection) {
-                $bIsTemplateEngine = (TCMSUser::CMSUserDefined() && 'true' == $bIsModuleChooser);
-                $bAllowResourceCollection = ($bAllowResourceCollection && !$bIsTemplateEngine); // not in cms template engine
-            }
+        if (false === $this->IsEnabled()) {
+            return false;
+        }
+
+        $oGlobal = TGlobal::instance();
+        $bIsModuleChooser = $oGlobal->GetUserData('__modulechooser');
+        if (true === TGlobal::IsCMSMode()) {
+            // do not enable in backend mode
+            return false;
+        }
+        $bIsTemplateEngine = 'true' === $bIsModuleChooser && TCMSUser::CMSUserDefined();
+        if (true === $bIsTemplateEngine) {
+            return false;
         }
 
-        return $bAllowResourceCollection;
+        if (true === $this->resourceCollectionProcessRunning()) {
```

Comment:
-> "isRunning()" ?

Reply:
isRunning is not quite correct - it is more a "is currently writing" info. So renamed methods to

hasResourceCollectionWriteProcessRunning

and 

setsResourceCollectionWriteProcessRunning